plugin/file/file.go

```diff
@@ -121,6 +121,12 @@ func (s *serialErr) Error() string {
 // it returns an error indicating nothing was read.
 func Parse(f io.Reader, origin, fileName string, serial int64) (*Zone, error) {
 	tokens := dns.ParseZone(f, dns.Fqdn(origin), fileName)
+	defer func() {
```

Comment:
This looks like a proper fix, but it sure makes for a messy API. Wonder if something can be done in dns.ParseZone itself.